homeassistant/helpers/service.py

```diff
@@ -99,6 +100,18 @@ def extract_entity_ids(hass, service_call):
 
     Will convert group entity ids to the entity ids it represents.
     """
+    return run_callback_threadsafe(
+        hass.loop, async_extract_entity_ids, hass, service_call
+    ).result()
+
+
+def async_extract_entity_ids(hass, service_call):
```

Comment:
I don't think this method is necessary since all methods it calls are both async and thread friendly (hence no code changes are recorded)